src/Compiler/Driver/fsc.fs

```diff
@@ -473,6 +551,8 @@ let main1
         disposables: DisposablesTracker
     ) =
 
+    CompilerGlobalState.stampCount <- 0L
```

Comment:
Horrible hack to keep stamps in tact... instead, the whole state should be cached at the right moment and restored along with the stamps.